File: src/main/java/pl/kul/inpost_assessment/model/parcelLocker/ParcelLocker.java
package pl.kul.inpost_assessment.model.parcelLocker;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class ParcelLocker {
    @Id
    private String name;
    private String locationDescription;

    @Enumerated(EnumType.STRING)
    private CountryCode country;

    @Enumerated(EnumType.STRING)
    private ParcelLockerStatus status;
    private double longitude;
    private double latitude;

    @Embedded
    private Address address;
}

File: frontend/src/app/model/ParcelLocker.ts
import {ParcelLockerStatus} from './unionTypes/ParcelLockerStatus';
import {CountryCode} from './unionTypes/CountryCode';
import {Address} from './Address';

export interface ParcelLocker {
    name: string,
    locationDescription: string,
    country: CountryCode,
    status: ParcelLockerStatus,
    longitude: number,
    latitude: number
    address: Address
}
